src/lib.rs

```diff
@@ -11,18 +11,17 @@ pub struct System {
 }
 
 #[derive(Debug)]
-pub struct MavsdkError<ErrT> {
-    pub result: ErrT,
-    pub result_str: String,
+pub enum RequestError<PluginMavErr> {
+    MavErr(PluginMavErr),
+    RpcErr(grpcio::Error),
 }
 
-#[derive(Debug)]
-pub enum MavsdkResult<T, ErrT> {
-    Ok(T),
-    Err(MavsdkError<ErrT>),
-    RpcErr(grpcio::Error),
+trait FromRpcResult<T> {
+    fn from_rpc_result(rpc_result: ::grpcio::Result<T>) -> Self;
 }
 
+pub type RequestResult<SuccessType, PluginMavErr> = Result<SuccessType, RequestError<PluginMavErr>>;
```

Comment:
@JonasVautherin what are you think about that?

Reply:
Is that synctactic sugar abstracting away `RequestError`? Fancy :sunglasses:.

Reply:
Yes, it’s hard not to use such nice things :)